File: DotNet/CesiumLanguageWriter/CameraCesiumWriter.cs
using System;
#if StkComponents
using AGI.Foundation.Cesium.Advanced;
#else
using CesiumLanguageWriter.Advanced;
#endif

#if StkComponents
namespace AGI.Foundation.Cesium
#else
namespace CesiumLanguageWriter
#endif
{
    /// <summary>
    /// A <see cref="CesiumPropertyWriter{TDerived}"/> for writing camera-related properties to a
    /// <topic name="Cesium">Cesium</topic> stream. The camera is positioned in the scene
    /// by the <see cref="CesiumPacketWriter.OpenPositionProperty">Position</see> property.  Instances of this class should not be created
    /// directly, but should instead be obtained from a <see cref="CesiumPacketWriter"/>.
    /// </summary>
    public class CameraCesiumWriter : CesiumPropertyWriter<CameraCesiumWriter>
    {
        /// <summary>
        /// The name of the <code>enable</code> property.
        /// </summary>
        public static readonly string EnablePropertyName = "enable";

        private readonly Lazy<BooleanCesiumWriter> m_enable = new Lazy<BooleanCesiumWriter>(() => new BooleanCesiumWriter(EnablePropertyName), false);
        /// <summary>
        /// Initializes a new instance.
        /// </summary>
        public CameraCesiumWriter(string propertyName)
            : base(propertyName)
        {
        }

        /// <summary>
        /// Initializes a new instance as a copy of an existing instance.
        /// </summary>
        /// <param name="existingInstance">The existing instance to copy.</param>
        protected CameraCesiumWriter(CameraCesiumWriter existingInstance)
            : base(existingInstance)
        {
        }

        /// <inheritdoc />
        public override CameraCesiumWriter Clone()
        {
            return new CameraCesiumWriter(this);
        }

        /// <summary>
        /// Gets the writer for the <code>enable</code> property.  The returned instance must be opened by calling
        /// the <see cref="CesiumElementWriter.Open"/> method before it can be used for writing.  See the
        /// documentation for the <see cref="OpenEnableProperty"/> method for more information.
        /// </summary>
        public BooleanCesiumWriter EnableWriter
        {
            get { return m_enable.Value; }
        }

        /// <summary>
        /// Opens a property describing whether the camera is enabled. 
        /// </summary>
        /// <returns>The writer that is used to write color information.</returns>
        public BooleanCesiumWriter OpenEnableProperty()
        {
            OpenIntervalIfNecessary();
            return OpenAndReturn(EnableWriter);
        }

        /// <summary>
        /// Writes a constant value for the <code>enable</code> property.
        /// See the documentation for the <see cref="OpenEnableProperty"/> method for more information.
        /// </summary>
        /// <param name="value">The value of the property.</param>
        public void WriteEnableProperty(bool value)
        {
            using (var writer = OpenEnableProperty())
            {
                writer.WriteValue(value);
            }
        }
    }
}

File: Java/CesiumLanguageWriter/translatedSrc/cesiumlanguagewriter/CameraCesiumWriter.java
package cesiumlanguagewriter;


import agi.foundation.compatibility.*;
import agi.foundation.compatibility.DisposeHelper;
import agi.foundation.compatibility.Func1;
import agi.foundation.compatibility.Lazy;
import cesiumlanguagewriter.advanced.*;

/**
 *  
 Writes a <code>Camera</code> to a  {@link CesiumOutputStream}.  A <code>Camera</code> a camera.
 

 */
public class CameraCesiumWriter extends CesiumPropertyWriter<CameraCesiumWriter> {
	/**
	 *  
	The name of the <code>enable</code> property.
	

	 */
	public static final String EnablePropertyName = "enable";
	private Lazy<BooleanCesiumWriter> m_enable = new Lazy<BooleanCesiumWriter>(new Func1<BooleanCesiumWriter>() {
		public BooleanCesiumWriter invoke() {
			return new BooleanCesiumWriter(EnablePropertyName);
		}
	}, false);

	/**
	 *  
	Initializes a new instance.
	

	 */
	public CameraCesiumWriter(String propertyName) {
		super(propertyName);
	}

	/**
	 *  
	Initializes a new instance as a copy of an existing instance.
	
	

	 * @param existingInstance The existing instance to copy.
	 */
	protected CameraCesiumWriter(CameraCesiumWriter existingInstance) {
		super(existingInstance);
	}

	@Override
	public CameraCesiumWriter clone() {
		return new CameraCesiumWriter(this);
	}

	/**
	 *  Gets the writer for the <code>enable</code> property.  The returned instance must be opened by calling the  {@link CesiumElementWriter#open} method before it can be used for writing.  The <code>enable</code> property defines whether or not the camera is enabled.
	

	 */
	public final BooleanCesiumWriter getEnableWriter() {
		return m_enable.getValue();
	}

	/**
	 *  
	Opens and returns the writer for the <code>enable</code> property.  The <code>enable</code> property defines whether or not the camera is enabled.
	

	 */
	public final BooleanCesiumWriter openEnableProperty() {
		openIntervalIfNecessary();
		return this.<BooleanCesiumWriter> openAndReturn(getEnableWriter());
	}

	/**
	 *  
	Writes a value for the <code>enable</code> property as a <code>boolean</code> value.  The <code>enable</code> property specifies whether or not the camera is enabled.
	
	

	 * @param value The value.
	 */
	public final void writeEnableProperty(boolean value) {
		{
			cesiumlanguagewriter.BooleanCesiumWriter writer = openEnableProperty();
			try {
				writer.writeBoolean(value);
			} finally {
				DisposeHelper.dispose(writer);
			}
		}
	}
}